ivy/functional/ivy/data_type.py

```diff
@@ -2127,28 +2123,24 @@ def promote_types(
         The type that both input types promote to
     """
     query = [ivy.as_ivy_dtype(type1), ivy.as_ivy_dtype(type2)]
-    query.sort(key=lambda x: str(x))
     query = tuple(query)
+    if query not in ivy.promotion_table:
+        query = (query[1], query[0])
 
     def _promote(query):
         if array_api_promotion:
             return ivy.array_api_promotion_table[query]
         return ivy.promotion_table[query]
 
-    try:
-        ret = _promote(query)
-    except KeyError:
-        # try again with the dtypes swapped
-        query = (query[1], query[0])
-        try:
-            ret = _promote(query)
-        except KeyError:
-            raise ivy.utils.exceptions.IvyDtypePromotionError(
-                "these dtypes ({} and {}) are not type promotable, ".format(
-                    type1, type2
-                )
-            )
-    return ret
+    # try:
```

Comment:
let's remove the commented out code. Also, could you please confirm (by just looping over all pairs) to see if all pairs of dtypes are covered in the promotion tables either forward or reverse (both with the precision and non-precision modes)? Thanks!

Reply:
You can check with and without the precision mode by once running the loop with `ivy.set_precision_mode(True)` and once with `ivy.set_precision_mode(False)` globally

Reply:
Yes I can confirm that all pairs of dtypes are covered in the promotion table. However I use the `ivy.set_precision_mode()` because it threw an error `AttributeError: module 'ivy' has no attribute 'set_precision_mode'`

Reply:
Oh sorry, I made a mistake here, it should be `ivy.set_precise_mode` 😅